### ISSUE TITLE
pnpm build fails with "cannot copy to non-directory"

### ISSUE THREAD

**Issue by frontsideair** [railwayapp/nixpacks#252]
The COPY command after `pnpm install` fails with "cannot copy to non-directory".

After googling a bit I found out that this error is related to copying over a symlink, and pnpm works with symbolic links.

I encountered this in my fairly complex project, so I reproduced it with a simple example with a handful of dependencies. (`react` and `react-dom` only)

Using the latest nixpacks@0.1.7 on macOS aarch64.

<details>
<summary>Logs:</summary>

```
% nixpacks build .  
=== Building (nixpacks v0.1.7) ===
=> Packages
    -> nodejs
    -> nodePackages.pnpm
=> Install
    -> pnpm i --frozen-lockfile
=> Build
    -> Skipping
=> Start
    -> Skipping
[+] Building 5.0s (11/11) FINISHED                                                                                           
 => [internal] load build definition from Dockerfile                                                                    0.0s
 => => transferring dockerfile: 405B                                                                                    0.0s
 => [internal] load .dockerignore                                                                                       0.0s
 => => transferring context: 2B                                                                                         0.0s
 => [internal] load metadata for ghcr.io/railwayapp/nixpacks:debian-1655126806                                          0.5s
 => [internal] load build context                                                                                       2.0s
 => => transferring context: 63.35MB                                                                                    2.0s
 => [1/7] FROM ghcr.io/railwayapp/nixpacks:debian-1655126806@sha256:3d33b3f942fff25f99270e78177fe5ea39ce94b4090c5af336  0.0s
 => CACHED [2/7] WORKDIR /app/                                                                                          0.0s
 => CACHED [3/7] COPY environment.nix /app/                                                                             0.0s
 => CACHED [4/7] RUN nix-env -if environment.nix                                                                        0.0s
 => [5/7] COPY . /app/                                                                                                  0.4s
 => [6/7] RUN pnpm i --frozen-lockfile                                                                                  1.9s
 => ERROR [7/7] COPY . /app/                                                                                            0.1s
------                                                                                                                       
 > [7/7] COPY . /app/:                                                                                                       
------                                                                                                                       
cannot copy to non-directory: /var/lib/docker/overlay2/jkgrgt105arukauuwwjunqb5w/merged/app/node_modules/.pnpm/loose-envify@1.4.0/node_modules/js-tokens                                                                                                  
Error: Docker build failed
```

</details>

**Comment by frontsideair** [comment]
I think it has to do with the fact my local directory has `node_modules` directory. I tried with the pnpm example and it worked fine until I installed dependencies locally.

I think the solution would be to add a `.dockerignore` file.

**Comment by wyzlle** [comment]
Fixed by #254
